test/integration_test.py

```diff
@@ -2062,17 +2071,31 @@ def test_response_security_headers(self):
                     else:
                         response = requests.get(str(endpoint / path))
                         response.raise_for_status()
-                        expected = AzulChaliceApp.security_headers | expected_headers
-                        # FIXME: Add a CSP header with a nonce value to text/html responses
-                        #        https://github.com/DataBiosphere/azul-private/issues/6
+                        expected = AzulChaliceApp.security_headers() | expected_headers
+                        csp = response.headers['Content-Security-Policy']
+                        # These paths have a CSP that differs from the default
+                        # by including randomly generated 'nonce' values.
                         if path in ['/', '/oauth2_redirect']:
-                            del expected['Content-Security-Policy']
+                            nonces = set()
+                            for directive in csp.split(';'):
+                                match = directive_re.fullmatch(directive)
+                                self.assertIsNotNone(match)
+                                name, value = match.groups()
+                                if name in ('script-src', 'style-src'):
+                                    match = nonce_re.search(value)
+                                    self.assertIsNotNone(match)
+                                    nonces.add(match.group(1))
+                            nonce = one(nonces)
+                            expected_csp = AzulChaliceApp.content_security_policy(nonce)
+                            expected['Content-Security-Policy'] = expected_csp
+                        else:
+                            self.assertNotIn('nonce', csp)
```

Comment:
I made it clear in an earlier review that we shouldn't use substring search.

You need to parse the CSP regardless of which endpoint is hit. Once you've parsed it, you can reliably assert the absence of nonces.